src/simulatorwebview/localserver.ts

```diff
@@ -0,0 +1,188 @@
+            res.setHeader("Access-Control-Allow-Credentials", "true");
+            if (req.method === "OPTIONS") {
+                res.end();
+            } else {
+                next();
+            }
+        });
+        this.app.use(bodyParser.json());
+        this.app.use(bodyParser.urlencoded({extended: false}));
+        this.app.use("/", this.router);
+        this.app.set("trust proxy", 1);
+        this.app.use("*", (req, res) => {
+            res.status(404).json({ error: "I don\'t have that" });
+        });
+        this.app.use("*", (err, req, res, next) => {
+            if (err) {
+              res.status(500).json({error: err.toString()});
+            } else {
+              res.status(404).json({error: "I don\'t have that"});
+            }
+        });
+    }
+
+    private async getPreSelected(req: express.Request, res: express.Response, next: express.NextFunction) {
+        try {
+            return res.status(200).json(this.preSelectedDevice);
+        } catch (err) {
+            next(err);
+        }
+    }
+
+    private async isProcessing(req: express.Request, res: express.Response, next: express.NextFunction) {
+        try {
+            const processing = await Simulator.isProcessing();
+            return res.status(200).json(processing);
+        } catch (err) {
+            next(err);
+        }
+    }
+
+    private async setProcessing(req: express.Request, res: express.Response, next: express.NextFunction) {
+        try {
+            const data = req.body;
+            const processing = data.processing;
+            Simulator.setProcessing(processing);
+            return res.status(200).json(processing);
+        } catch (err) {
+            next(err);
+        }
+    }
+
+    private async getIoTHubHostName(req: express.Request, res: express.Response, next: express.NextFunction) {
+        try {
+            const iotHubConnectionString = await Utility.getConnectionString(Constants.IotHubConnectionStringKey, Constants.IotHubConnectionStringTitle, false);
+            const result = ConnectionString.parse(iotHubConnectionString).HostName;
+            return res.status(200).json(result);
+        } catch (err) {
+            next(err);
+        }
+    }
+
+    private async getInputDeviceList(req: express.Request, res: express.Response, next: express.NextFunction) {
+        try {
+            const list = await Simulator.getInputDeviceList();
+            return res.status(200).json(list);
+        } catch (err) {
+            next(err);
+        }
+    }
+
+    private async send(req: express.Request, res: express.Response, next: express.NextFunction) {
+        try {
+            // Since we do not support C2D yet, here is just one call on sendD2C()
+            await this.sendD2C(req, res, next);
+            // Must return a status here. If no success or failure returned, the webview may retry and cause unexpected re-send behavior.
+            res.sendStatus(200); 
```

Comment:
remove trailing whitespace. Need to run tslint locally, or better to just run it in Travis CI.